packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -1275,19 +1357,45 @@ function beginWork(
       return updateContextProvider(
         current,
         workInProgress,
-        renderExpirationTime,
+        renderExpirationTime
       );
     case ContextConsumer:
       return updateContextConsumer(
         current,
         workInProgress,
-        renderExpirationTime,
+        renderExpirationTime
+      );
+    case PureComponent: {
+      const type = workInProgress.type;
+      return updatePureComponent(
+        current,
+        workInProgress,
+        type,
+        workInProgress.pendingProps,
+        updateExpirationTime,
+        renderExpirationTime
       );
+    }
+    case PureComponentLazy: {
```

Comment:
this duplication kills me a little bit :)

Reply:
Me too :D